commands.js

```diff
@@ -1044,16 +1044,48 @@ exports.commands = {
 			move = Commands.randommove.call(this, "2,viable", self, dummyRoom);
 		}
 
+		//Code for generating a random name based on types generated
+		//Start by accessing the array from the capnamelist.js file in the data folder.
+		var foundName = false;
+		var t1Name = '';
+		var type1array = Data.Capnames[type[0]];
+		var type2array = Data.Capnames[type[1]];
```

Comment:
There are two errors here:

The first is that the `type` array holds the type names capitalized, which the Capnames table holds keys of all lowercase. This can be fixed by either changing the keys to be capitalized as well, or by using `.toLowerCase()` on the type. The former might be easier depending on how you want to approach the second problem.

Since this can generate single typed pokemon it's possible that `type[1]` is undefined. I'm not sure how you want to handle situations of single typed pokemon, but it might be easiest to check if `type.length === 2`` and then just set `type2array` = `type1array` if that is the case. If this doubling up the type approach is taken, it'll create a further issue of the possibility of the same name getting called twice and it repeating (i.e. "hornhorn" becomes a possible name). Since that is such a rare and minor issue, however, if you don't deem it worth addressing I'll except it anyways.